source/dagger/gameplay/team_game/golem.cpp

```diff
@@ -0,0 +1,180 @@
+#include "golem.h"
+
+#include "mage.h"
+#include "range_of_attack.h"
+
+#include "core/engine.h"
+#include "core/game/transforms.h"
+#include "core/graphics/animation.h"
+#include "core/graphics/sprite.h"
+
+#include "gameplay/common/simple_collisions.h"
+
+#include <math.h>
+#include <list>
+
+using namespace ancient_defenders;
+
+
+void ancient_defenders::GolemBehaviorSystem::SpinUp()
+{
+	Engine::Dispatcher().sink<NextFrame>().connect<&GolemBehaviorSystem::OnEndOfFrame>(this);
+}
+
+void ancient_defenders::GolemBehaviorSystem::WindDown()
+{
+	Engine::Dispatcher().sink<NextFrame>().disconnect<&GolemBehaviorSystem::OnEndOfFrame>(this);
+}
+
+void ancient_defenders::GolemBehaviorSystem::Run()
+{
+	Engine::Registry().view<GolemStats, Sprite, Transform, Animator, RangeOfAttack>().each(
+		[](GolemStats& golem_, Sprite& sprite_, Transform& transform_, Animator& animation_, RangeOfAttack& range_)
+		{
+			if (range_.targetFound) golem_.currentAction = EAction::Attacking;
+
+			if (golem_.currentAction == EAction::Idling) {
+				AnimatorPlay(animation_, "ancient_defenders:golem:IDLE");
+			}
+			else if (golem_.currentAction == EAction::Moving) {
+				auto nextPosition = golem_.postition - 1;
+
+				auto coords = WalkingPath::path.back();
+				int i = 0;
+				for (auto iter : WalkingPath::path) {
+					if (i == nextPosition) {
+						coords = iter;
+						break;
+					}
+					i++;
+				}
+
+				auto destinationX = coords.x; // Maybe add +/- a few percent to make paths bit more varied
+				auto destinationY = coords.y;
+
+				if (transform_.position.x < destinationX) {
+					golem_.direction.x = 1;
+				}
+				else if (transform_.position.x > destinationX) {
+					golem_.direction.x = -1;
+				}
+				else golem_.direction.x = 0;
+				if (transform_.position.y < destinationY) {
+					golem_.direction.y = 1;
+				}
+				else if (transform_.position.y > destinationY) {
+					golem_.direction.y = -1;
+				}
+				else golem_.direction.y = 0;
+
+				if (golem_.direction.x != 0) {
+					sprite_.scale.x = (-golem_.direction.x) * abs(sprite_.scale.x);
+					AnimatorPlay(animation_, "ancient_defenders:golem:WALK_SIDE");
+				}
```

Comment:
Since the golems don't have different animations for each moving direction it enough to just do AnimatorPlay without checking the directions since the same animation will be played.